ARrehab/ARrehab/MovementGame.swift

```diff
@@ -47,15 +54,49 @@ class MovementGame : Minigame {
         self.total = num
         // For our purposes, we placed the player as a 2 centimeter sphere around the camera.
         self.playerCollisionEntity = TriggerVolume(shape: ShapeResource.generateSphere(radius: 0.01), filter: CollisionFilter(group:playerCollisionGroup, mask: targetCollisionGroup))
+        self.coachingState = .other
         super.init()
+        self.progress = [0, 0]
+        self.timer = Timer.scheduledTimer(withTimeInterval: 0.01, repeats: true) { timer in
+            self.progress[1] = (-self.playerCollisionEntity.convert(position: SIMD3<Float>(0,0,0), to: self).y)/0.4
+            var stateIsDown : MovementState? = nil
+            self.children.forEach { (entity) in
+                guard let target = entity as? MovementTarget else {
+                    return
+                }
+                if (stateIsDown == nil) {
+                    stateIsDown = target.state
+                } else if (stateIsDown != target.state) {
+                    stateIsDown = .other
+                }
+            }
+            // Coach the next step
+            if (stateIsDown == .up) {
+                stateIsDown = .down
+            } else if stateIsDown == .down {
+                stateIsDown = .up
+            }
+            self.coachingState = stateIsDown ?? .other
+        }
+        self.timer.tolerance = 0.1
+        
         // Create a target with a trigger time of 1 second
-        let target = MovementTarget(delay: 1, reps: num)
+        let target = MovementTarget(delay: 1, reps: num, arrow: true)
         target.collision?.filter = CollisionFilter(group: self.targetCollisionGroup, mask: self.playerCollisionGroup)
         // Change the orientation to squating rather than to the left
         target.transform.rotation = simd_quatf(angle: -.pi/2, axis: SIMD3<Float>(0,0,1))
         // Move the squat target down by 0.2 m.
         target.transform.translation = SIMD3<Float>(0,-0.2,0)
         self.addChild(target)
+        
+        // Create a target with a trigger time of 1 second
+        let hardTarget = MovementTarget(delay: 1, reps: num, arrow: false)
+        hardTarget.collision?.filter = CollisionFilter(group: self.targetCollisionGroup, mask: self.playerCollisionGroup)
+        // Change the orientation to squating rather than to the left
+        hardTarget.transform.rotation = simd_quatf(angle: -.pi/2, axis: SIMD3<Float>(0,0,1))
```

Comment:
Ohhhh so is this how quaternions work?? Might need a little explanation here personally

Reply:
Honestly I have a hard time understanding how quaternions work but this is a constructor that creates a quaternion that represents a rotation about the axis by the angle.

But will add to the documentation.